control-plane/pkg/reconciler/trigger/controllerv2.go

```diff
@@ -0,0 +1,107 @@
+	brokerinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1/broker"
+	triggerinformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1/trigger"
+	triggerreconciler "knative.dev/eventing/pkg/client/injection/reconciler/eventing/v1/trigger"
+	eventinglisters "knative.dev/eventing/pkg/client/listers/eventing/v1"
+
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/config"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/kafka"
+
+	internalscg "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1"
+	consumergroupclient "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/client"
+	consumergroupinformer "knative.dev/eventing-kafka-broker/control-plane/pkg/client/internals/kafka/injection/informers/eventing/v1alpha1/consumergroup"
+)
+
+func NewControllerV2(ctx context.Context, configs *config.Env) *controller.Impl {
+
+	logger := logging.FromContext(ctx).Desugar()
+
+	brokerInformer := brokerinformer.Get(ctx)
+	triggerInformer := triggerinformer.Get(ctx)
+	triggerLister := triggerInformer.Lister()
+	consumerGroupInformer := consumergroupinformer.Get(ctx)
+
+	reconciler := &ReconcilerV2{
+		BrokerLister:        brokerInformer.Lister(),
+		EventingClient:      eventingclient.Get(ctx),
+		Env:                 configs,
+		ConsumerGroupLister: consumerGroupInformer.Lister(),
+		InternalsClient:     consumergroupclient.Get(ctx),
+	}
+
+	impl := triggerreconciler.NewImpl(ctx, reconciler, func(impl *controller.Impl) controller.Options {
+		return controller.Options{
+			FinalizerName:     FinalizerName,
+			AgentName:         ControllerAgentName,
+			SkipStatusUpdates: false,
+		}
+	})
+
+	triggerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: filterTriggers(reconciler.BrokerLister),
+		Handler:    controller.HandleAll(impl.Enqueue),
+	})
+
+	// Filter Brokers and enqueue associated Triggers
+	brokerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: kafka.BrokerClassFilter(),
+		Handler:    enqueueTriggers(logger, triggerLister, impl.Enqueue),
+	})
+
+	// ConsumerGroup changes and enqueue associated Trigger
+	consumerGroupInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		Handler: enqueueTrigger(logger, triggerLister, impl.Enqueue),
+	})
+
+	return impl
+}
+
+func enqueueTrigger(
+	logger *zap.Logger,
+	lister eventinglisters.TriggerLister,
+	enqueue func(obj interface{})) cache.ResourceEventHandler {
+
+	return controller.HandleAll(func(obj interface{}) {
+
+		if cg, ok := obj.(*internalscg.ConsumerGroup); ok {
+			triggers, err := lister.Triggers(cg.Namespace).List(labels.Everything())
+			if err != nil {
+				logger.Warn("Failed to list triggers", zap.Error(err))
+				return
+			}
+
+			for _, trigger := range triggers {
+				if cg.Name == string(trigger.UID) { //
+					enqueue(trigger)
+				}
+			}
+		}
+	})
```

Comment:
We can use the Trigger that is in the owenerreferences list, no need to list triggers, and then filtering.

Something like:
```golang
for _, or := range cg.OwnerReferences {
  if strings.ToLower(or.Kind) == "trigger" {
    impl.EnqueueKey(types.NamespaceName{Namespace: cg.GetNamespace(), Name: or.Name})
    return
  } 
}
```

Reply:
Great - that should work